### PR TITLE
feat(web): search referenced item by title value

### PR DESCRIPTION
**Pull request by caichi-t**
## What I've done
- Changed title of referenced items modal
- Fixed the modal style
- Added function that searching the items by Title of them

## Memo
Currently, the id of the Title column is got in a roundabout way, so it might be better to change it.

### CODE REVIEW

File: web/src/components/organisms/Project/Content/ContentList/hooks.ts [hunk 1]
```diff
@@ -334,6 +334,7 @@ export default () => {
   const handleSearchTerm = useCallback(
     (term?: string) => {
       searchParams.set("searchTerm", term ?? "");
+      searchParams.set("page", "1");
```
Comment (caichi-t) [inline web/src/components/organisms/Project/Content/ContentList/hooks.ts:337]: Fixed page returns to page 1 after searching on a page other than page 1 on main table.

File: web/src/components/molecules/Content/Form/ReferenceFormItem/index.tsx [hunk 2]
```diff
@@ -50,7 +54,10 @@ const ReferenceFormItem: React.FC<Props> = ({
     if (!onReferenceModelUpdate) return;
     onReferenceModelUpdate(modelId);
     setVisible(true);
-  }, [setVisible, onReferenceModelUpdate, modelId]);
+    if (onSearchTerm) {
+      onSearchTerm("");
+    }
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/Form/ReferenceFormItem/index.tsx:59]:
```suggestion
    onSearchTerm?.("");
```

File: web/src/components/molecules/Content/LinkItemModal/index.tsx [hunk 3]
```diff
@@ -99,55 +108,94 @@ const LinkItemModal: React.FC<Props> = ({
...
     },
   ];
 
+  useEffect(() => {
+    if (!visible) {
+      setValue("");
+    }
+  }, [visible]);
+
+  const onInput = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/LinkItemModal/index.tsx:140]:
```suggestion
  const handleInput = (e: ChangeEvent<HTMLInputElement>) => {
```

File: web/src/components/molecules/Content/LinkItemModal/index.tsx [hunk 4]
```diff
@@ -99,55 +108,94 @@ const LinkItemModal: React.FC<Props> = ({
...
+        placeholder={t("input search text")}
+        onSearch={(value: string) => {
+          if (value) {
+            onSearchTerm(value);
+          } else {
+            onSearchTerm();
+          }
+        }}
+        value={value}
+        onChange={onInput}
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/LinkItemModal/index.tsx:157]:
```suggestion
        onChange={handleInput}
```

File: web/src/components/organisms/Project/Content/ContentDetails/hooks.ts [hunk 5]
```diff
@@ -61,11 +63,13 @@ export default () => {
   const [collapsedModelMenu, collapseModelMenu] = useState(false);
   const [collapsedCommentsPanel, collapseCommentsPanel] = useState(true);
   const [requestModalShown, setRequestModalShown] = useState(false);
+  const [searchTerm, setSearchTerm] = useState<string>("");
```
Comment (nourbalaha) [inline web/src/components/organisms/Project/Content/ContentDetails/hooks.ts:66]:
```suggestion
  const [searchTerm, setSearchTerm] = useState("");
```

**Review by nourbalaha — CHANGES_REQUESTED** [review]

**Review by nourbalaha — APPROVED** [review]

**Pull request merged**